llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6870,12 +6870,25 @@ Constant *SPIRVToLLVM::buildShaderInOutMetadata(SPIRVType *bt, ShaderInOutDecora
     SPIRVWord blockXfbOffset = SPIRVID_INVALID;
     SPIRVWord xfbOffset = SPIRVID_INVALID;
 
-    // Do iteration to find the minimum member transform feedback offset as
-    // starting block transform feedback offset
-    for (auto memberIdx = 0; memberIdx < numMembers; ++memberIdx)
+    // Do iteration to deal with transform feedback buffer info
+    // Check if the structure member specifies transform feedback id or stride
+    // Enable transform feedback buffer if transform feedback offset is declared, and then
+    // find the minimum member transform feedback offset as starting block transform feedback offset
+    for (auto memberIdx = 0; memberIdx < numMembers; ++memberIdx) {
+      if ((bt->hasMemberDecorate(memberIdx, DecorationXfbBuffer, 0, &xfbBuffer)) && (inOutDec.IsXfb == false)) {
+        inOutDec.IsXfb = true;
```

Comment:
As I said, we should set IsXfb to TRUE once we detect `xfb_offset` qualifier. In other cases, this flag should not be set, such as the `xfb_buffer` and `xfb_stride` are specified. The GLSL spec says: only variables with explicit offsets are captured.

Reply:
Sorry, I didn't notice SPIR-V allows `offset` decoration on normal structure even if transform feedback is not requested. In such case, we have to resort to xfb_buffer and xfb_stride. SPIR-V reuses `offset` to express xfb_offset. Which is inconvenient for us.